site.go

```diff
@@ -182,14 +185,26 @@ func (s *Site) GetS3Photo(key string) *S3Photo {
 	}
 }
 
-func (s *Site) GetImgixPhoto(key string) *ImgixPhoto {
+func (s *Site) GetScaledPhoto(resizingService string, resizingServiceSecret string, key string) Renderable {
 	if baseUrl, err := url.Parse(s.BaseUrl); err != nil {
 		fmt.Printf("Error trying to parse site base URL. Error: %s\n", err.Error())
 		return nil
 	} else {
-		return &ImgixPhoto{
-			key,
-			baseUrl,
+		if resizingService == "imgix" {
+			return &ImgixRescaledPhoto{
+				key,
+				baseUrl,
+			}
+		} else if resizingService == "thumbor" {
+			return &ThumborRescaledPhoto{
+				resizingServiceSecret,
+				key,
+				baseUrl,
+			}
+		} else {
+			//TODO: get better handling on this, maybe validate the config as it's being read?
```

Comment:
would you be interested in seeing this as an issue or is it not something that concerns you?

Reply:
Right now, the strategy that 50mm uses is to validate all config at startup. I like how with nginx, apache, etc you always start with a good config, or fail to start if config is bad. Much easier to get things right.

There should be existing functions that validate configs. You can put checks on the type of service used there, and fail if it's an unknown service. Ideally once the server is up and running, it should not fall down unless it's something out of our control.